lib/crowbar/client/command/proposal/edit.rb

```diff
@@ -140,6 +150,29 @@ def payload_content
               from_editor
             end
           end
+
+          def check_nodes(configuration)
+            avaliable_nodes = JSON.parse(Request::Node::List.new({}).process)
+            if avaliable_nodes["nodes"].nil?
+              say "Internal error: Invalid node list provided"
+              raise
+            end
+
+            avaliable_nodes = avaliable_nodes["nodes"]
+
+            configuration["deployment"].each do |service_name, service|
+              next if service["elements"].nil?
+
+              service["elements"].each do |role, nodes|
+                nodes.each do |node|
+                  if avaliable_nodes.select { |item| item["name"] == node }.empty?
+                    say "ERROR : Node #{node} is not available or correct. " \
```

Comment:
What does correct mean in a node context?

Reply:
A node is not available when it isn't in the list of possible nodes, and I assume that is not correct if this is an invalid IP.

Reply:
Alright, I would like to get this clarified in the error message. `not correct` can mean a lot of things. ;)

Reply:
Ok ,thanks